src/Polly.Specs/RateLimit/AsyncSlidingWindowRateLimiterTests.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using FluentAssertions;
+using Polly.RateLimit;
+using Polly.Utilities;
+using Xunit;
+
+namespace Polly.Specs.RateLimit;
+
+public class AsyncSlidingWindowRateLimiterTests : SlidingWindowRateLimiterTestsBase, IDisposable
+{
+    protected override IRateLimitPolicy GetPolicyViaSyntax(int numberOfExecutions, TimeSpan perTimeSpan)
+    {
+        return Policy.RateLimitAsync(numberOfExecutions, perTimeSpan, spreadUniformly:false);
+    }
+    
+    public void Dispose()
+    {
+        SystemClock.Reset();
+    }
+    
+    [Fact]
+    public void Syntax_should_throw_for_perTimeSpan_zero()
+    {
+        Action invalidSyntax = () => GetPolicyViaSyntax(1, TimeSpan.Zero);
+
+        invalidSyntax.Should().Throw<ArgumentOutOfRangeException>().And.ParamName.Should().Be("perTimeSpan");
+    }
+
+    [Fact]
+    public void Syntax_should_throw_for_perTimeSpan_infinite()
+    {
+        Action invalidSyntax = () => GetPolicyViaSyntax(1, System.Threading.Timeout.InfiniteTimeSpan);
+
+        invalidSyntax.Should().Throw<ArgumentOutOfRangeException>().And.ParamName.Should().Be("perTimeSpan");
+    }
+
+    [Fact]
+    public void Syntax_should_throw_for_numberOfExecutions_negative()
+    {
+        Action invalidSyntax = () => GetPolicyViaSyntax(-1, TimeSpan.FromSeconds(1));
+
+        invalidSyntax.Should().Throw<ArgumentOutOfRangeException>().And.ParamName.Should().Be("numberOfExecutions");
+    }
+
+    [Fact]
+    public void Syntax_should_throw_for_numberOfExecutions_zero()
+    {
+        Action invalidSyntax = () => GetPolicyViaSyntax(0, TimeSpan.FromSeconds(1));
+
+        invalidSyntax.Should().Throw<ArgumentOutOfRangeException>().And.ParamName.Should().Be("numberOfExecutions");
+    }
+
+    [Fact]
+    public void Syntax_should_throw_for_perTimeSpan_negative()
+    {
+        Action invalidSyntax = () => GetPolicyViaSyntax(1, TimeSpan.FromTicks(-1));
+
+        invalidSyntax.Should().Throw<ArgumentOutOfRangeException>().And.ParamName.Should().Be("perTimeSpan");
+    }
+
+    [Fact]
+    public async Task ShouldAllowAllExecutionsWithinThePolicyTimeSpanAndLimit()
+    {
+        FixClock();
+            
+        // 60 executions in 1 minute
+        var rateLimiter = (AsyncRateLimitPolicy) GetPolicyViaSyntax(60, new TimeSpan(0, 1, 0));
```

Comment:
I find this style more readable. It would be good to change all the usages of this in the files you're updating.

```suggestion
        var rateLimiter = (AsyncRateLimitPolicy) GetPolicyViaSyntax(60, TimeSpan.FromMinutes(1));
```

Reply:
In fact these three tests look like they could be consolidated into a single data driven test.